api/src/pcapi/core/users/email/update.py

```diff
@@ -191,6 +230,20 @@ def generate_email_change_confirmation_token(user: models.User, new_email: str,
     return token
 
 
+def generate_email_change_validation_token(user: models.User, new_email: str, expiration_date: datetime) -> str:
+    token = encode_jwt_payload({"current_email": user.email, "new_email": new_email}, expiration_date)
+    app.redis_client.set(get_validation_token_key(user), token)  # type: ignore [attr-defined]
+    app.redis_client.expireat(get_validation_token_key(user), expiration_date)  # type: ignore [attr-defined]
+    return token
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**💭 thought:** ‏On fait exactement le même type de logique dans `generate_email_change_confirmation_token`
On doit pouvoir factoriser cette logique